examples/slack-connector/src/main/java/com/salesforce/connector/slack/service/SlackService.java

```diff
@@ -0,0 +1,187 @@
+package com.salesforce.connector.slack.service;
+
+import com.salesforce.connector.chatbot.service.ChatbotService;
+import com.salesforce.connector.chatbot.service.SessionIdProvider;
+import com.salesforce.einsteinbot.sdk.client.model.BotResponse;
+import com.salesforce.einsteinbot.sdk.model.AnyResponseMessage;
+import com.salesforce.einsteinbot.sdk.model.ChoicesResponseMessage;
+import com.salesforce.einsteinbot.sdk.model.ResponseEnvelope;
+import com.salesforce.einsteinbot.sdk.model.SessionEndedResponseMessage;
+import com.salesforce.einsteinbot.sdk.model.TextResponseMessage;
+import com.google.common.base.Preconditions;
+import com.slack.api.model.block.composition.OptionObject;
+import com.slack.api.model.block.element.BlockElement;
+import com.salesforce.connector.slack.Constants;
+import com.slack.api.bolt.App;
+import com.slack.api.bolt.context.Context;
+import com.slack.api.model.block.ActionsBlock;
+import com.slack.api.model.block.LayoutBlock;
+import com.slack.api.model.block.composition.PlainTextObject;
+import com.slack.api.model.block.element.ButtonElement;
+import com.slack.api.model.block.element.StaticSelectElement;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+
+@Service
+public class SlackService {
+
+  private static final Logger logger = LoggerFactory.getLogger(SlackService.class);
+
+  @Autowired
+  private ChatbotService chatbotService;
+
+  private String replyInThreadChannels;
+
+  @PostConstruct
+  public void initialize(){
+
+    if(System.getenv(Constants.REPLY_IN_THREAD_CHANNELS)!=null){
+      this.replyInThreadChannels = System.getenv(Constants.REPLY_IN_THREAD_CHANNELS);
+    } else {
+      this.replyInThreadChannels = "";
+    }
+  }
+
+  public void addThreadToReplyInThread(String channelId){
+    if(this.replyInThreadChannels.contains(channelId)){
+      this.replyInThreadChannels = this.replyInThreadChannels.replace(channelId,
+          Constants.EMPTY_STRING);
+    }else {
+      this.replyInThreadChannels += "," + channelId;
+    }
+  }
+
+  public boolean isFlag(String key){
+    return (System.getenv(key)!=null && System.getenv(key).equalsIgnoreCase("true"));
+  }
+
+  public boolean isReplyInThreadChannel(String channel){
+    return (this.replyInThreadChannels!=null && this.replyInThreadChannels.contains(channel));
+  }
+
+  public String cleanTextResponse(String responseText, String user){
+    return responseText.replace(Constants.BOT_RESPONSE_CLEANUP_MATCH,"<@"+ user + ">");
+  }
+
+  public String cleanTextRequest(String text, boolean isAppMention){
+    String response = text;
+    if (isAppMention) {
+      response = response.substring(14);
+    }
+    return response.trim();
+  }
+
+  private void sendToSlack(BotResponse botResponse, Context ctx, String user, String channel,
```

Comment:
The method is too long. Can we break it up into smaller methods?

Reply:
Broken into smaller methods